acme/src/acme/client.py

```diff
@@ -133,12 +134,20 @@ def new_order(self, csr_pem: bytes, profile: Optional[str] = None) -> messages.O
         else:
             ipNames = san_ext.value.get_values_for_type(x509.IPAddress)
         identifiers = []
-        for name in dnsNames:
-            identifiers.append(messages.Identifier(typ=messages.IDENTIFIER_FQDN,
-                value=name))
+        added_ips = set()
         for ip in ipNames:
-            identifiers.append(messages.Identifier(typ=messages.IDENTIFIER_IP,
-                value=str(ip)))
+            ip_str = str(ip)
+            identifiers.append(messages.Identifier(typ=messages.IDENTIFIER_IP, value=ip_str))
+            added_ips.add(ip_str)
+        for name in dnsNames:
+            try:
+                ip_addr = ipaddress.ip_address(name)
+                ip_str = str(ip_addr)
+                if ip_str not in added_ips:
+                    identifiers.append(messages.Identifier(typ=messages.IDENTIFIER_IP, value=ip_str))
+                    added_ips.add(ip_str)
+            except ValueError:
+                identifiers.append(messages.Identifier(typ=messages.IDENTIFIER_FQDN, value=name))
```

Comment:
It looks like this code is trying to guess if an identifier from Subject CN or SAN DNS is actually an IP address. I would rather be explicit, and only treat identifiers from SAN IPAddresses as IP addresses. When it comes to the Subject CN, we should only interpret it as a DNS name.

Given that, I _think_ the existing code in this file should just work, and this diff can be reverted. Let me know if you think otherwise.